spring-boot/admin/src/main/java/eu/coatrack/admin/controllers/GatewayApiController.java

```diff
@@ -51,4 +65,27 @@ public ApiKey findApiKeyEntityByApiKeyValue(@RequestParam("keyValue") String api
     public ServiceApi findServiceByApiKeyValue(@RequestParam("apiKeyValue") String apiKeyValue) {
         return serviceApiRepository.findByApiKeyValue(apiKeyValue);
     }
+
+    @GetMapping( "/api/gateways/{gatewayId}/apiKeys")
+    public ResponseEntity<List<ApiKey>> findApiKeyListByGatewayId(@PathVariable("gatewayId") String gatewayId) {
+        Proxy proxy = null;
+        List<ApiKey> apiKeyList;
+        String apiKeyListCreationFailedMessage = "Creation of API key list for requesting gateway failed. " +
+                "This concerns the gateway with the ID {}. ";
+        try {
+            proxy = proxyRepository.findById(gatewayId);
+            apiKeyList = proxy.getServiceApis().stream().flatMap(serviceApi -> serviceApi.getApiKeys()
+                    .stream()).collect(Collectors.toList());
+            return new ResponseEntity<>(apiKeyList, HttpStatus.OK);
+        } catch (NullPointerException e){
```

Comment:
A `NullPointerException` can happen in many places, and most of the time it will indicate an "internal server error" rather than "404 not found".

Reply:
That is refactored now. Now a more general Exception is used as well as an internal server error as HttpStatus for the ResponseEntity.